test/index.spec.js

```diff
@@ -93,81 +83,79 @@ describe('datastore-pubsub', function () {
   let serializedRecord = null
 
   // prepare Record
-  beforeEach(function (done) {
+  beforeEach(() => {
     keyRef = `key${testCounter}`
     key = (new Key(keyRef)).toBuffer()
     record = new Record(key, Buffer.from(value))
 
     serializedRecord = record.serialize()
-    done()
   })
 
-  afterEach(function (done) {
+  afterEach(() => {
     ++testCounter
-    done()
   })
 
-  after(function (done) {
-    parallel([
-      (cb) => stopDaemon(ipfsdA, cb),
-      (cb) => stopDaemon(ipfsdB, cb)
-    ], done)
+  after(() => {
+    return Promise.all([
+      stopDaemon(ipfsdA),
+      stopDaemon(ipfsdB)
+    ])
   })
 
-  it('should subscribe the topic, but receive error as no entry is stored locally', function (done) {
+  it('should subscribe the topic, but receive error as no entry is stored locally', async () => {
     const dsPubsubA = new DatastorePubsub(pubsubA, datastoreA, peerIdA, smoothValidator)
     const subsTopic = keyToTopic(`/${keyRef}`)
 
-    pubsubA.ls((err, res) => {
-      expect(err).to.not.exist()
-      expect(res).to.exist()
-      expect(res).to.not.include(subsTopic) // not subscribed key reference yet
-
-      dsPubsubA.get(key, (err) => {
-        expect(err).to.exist() // not locally stored record
-        expect(err.code).to.equal('ERR_NOT_FOUND')
-
-        pubsubA.ls((err, res) => {
-          expect(err).to.not.exist()
-          expect(res).to.exist()
-          expect(res).to.include(subsTopic) // subscribed key reference
-          done()
-        })
-      })
-    })
+    let subscribers = await pubsubA.ls()
+
+    expect(subscribers).to.exist()
+    expect(subscribers).to.not.include(subsTopic) // not subscribed key reference yet
+
+    let res
+    try {
+      res = await dsPubsubA.get(key)
+    } catch (err) {
+      expect(err).to.exist() // not locally stored record
+      expect(err.code).to.equal('ERR_NOT_FOUND')
+    }
+    expect(res).to.not.exist()
```

Comment:
The `catch` here is not guaranteed to be called. It's possible a bug could exist causing `.get` to not throw but return null or undefined. This could cause the subsequent expects to pass.

I would everything from `let res` to expecting it not to be defined with:
```js
await dsPubsubA.get(key).then(expect.fail, (err) => {
  expect(err).to.exist() // not locally stored record
  expect(err.code).to.equal('ERR_NOT_FOUND')
})
```

This guarantees that if `get` does not throw, the test will fail.